src/controllers/url-controller.ts

```diff
@@ -0,0 +1,152 @@
+import { Request, Response } from 'express';
+import prisma from '../libs/prisma-client';
+
+import { convertToHyphenated, generateShortUrl } from '../utils/short-url-generator';
+import logger from '../logs/logger';
+
+const successMsg = 'Url successfully fetched';
+const notFoundMsg = 'No Url found!';
+
+//  CREATE A SHORTEN URL FROM A LONG URL
+export const createUrl = async (req: Request, res: Response) => {
+  const { url, customName } = req.body;
+
+  if (!url) return res.status(400).send({ message: 'Url is required!' });
+  try {
+    const existingUrl = await prisma.url.findFirst({
+      where: {
+        originalUrl: url,
+      },
+    });
+    if (existingUrl) {
+      return res.status(400).send({ message: 'Url already exists!' });
+    }
+
+    const urlString = generateShortUrl();
+
+    const custom = customName && convertToHyphenated(customName);
```

Comment:
I could not find anywhere you validated that `customName` is a string with at least 5 letters. Move that validation to the middleware

Reply:
Custom name validation done.

Reply:
done